velox/dwio/common/ScanSpec.cpp

```diff
@@ -19,22 +19,25 @@
 
 namespace facebook::velox::common {
 
+ScanSpec* ScanSpec::getOrCreateChild(const std::string& name) {
+  if (auto it = this->childByFieldName_.find(name);
+      it != this->childByFieldName_.end()) {
+    return it->second;
+  }
+  this->children_.push_back(std::make_unique<ScanSpec>(name));
+  auto* child = this->children_.back().get();
+  this->childByFieldName_[child->fieldName()] = child;
+  return child;
+}
+
 ScanSpec* ScanSpec::getOrCreateChild(const Subfield& subfield) {
```

Comment:
@Yuhta Thank you for proposing to add `getOrCreateChild(const std::string& name)`. I verified it in Gluten and it can solve the tokenzier issue.

Still, I'm not sure I understand the typical usage of `getOrCreateChild(const Subfield& subfield)`. I wonder if it's appropriate to substitute `getOrCreateChild(const std::string& name)` for every string-to-subfield call? I see that there is one left in the runtime code and more in the tests. Would you help provide more context? Thanks.


https://github.com/facebookincubator/velox/blob/f34073472c2a7562cb89360e7c9891cc1a0b5ec8/velox/connectors/hive/iceberg/PositionalDeleteFileReader.cpp#L62-L63

Reply:
Found `getOrCreateChild(const Subfield&)` was still used by the runtime code so I keep it here. Please let me know if we need to make it test-only in some way. Thanks.
https://github.com/facebookincubator/velox/blob/d805d31a081ff593bf50b3f52775ebad5876354e/velox/dwio/common/MetadataFilter.cpp#L47

Reply:
This one is ok, the source of `field_` here is from a `ITypedExpr` so not related to the parser